script/check-size

```diff
@@ -120,11 +120,13 @@ size_nrf52840()
         "JOINER=1"
         "LINK_RAW=1"
         "MAC_FILTER=1"
+        "MLR=1"
         "MTD_NETDIAG=1"
         "SERVICE=1"
         "SLAAC=1"
         "SNTP_CLIENT=1"
         "TIME_SYNC=1"
+        "THREAD_VERSION=1.2"
```

Comment:
Continuing the discussion from https://github.com/openthread/openthread/pull/5241#discussion_r455884634:

> I was hoping to catch situations where adding Thread 1.2 code (intentionally or unintentionally) changes Thread 1.1 code. Should we flag situations where adding a Thread 1.2 feature significantly increases Thread 1.1 code?

Reply:
I fully agree with this. Having a 1.2 version size vs  1.1 version would be useful (to see the impact of 1.2 features on 1.1 code size).

Reply:
But it's beyond the scope of this PR. Shall we just turn off size check for this PR?

Reply:
I agree this should be in a separate PR. Can we take a look at what it would take to support size reports for both 1.1 and 1.2?

/cc/ @bukepo @librasungirl

Reply:
One thing I can imagine is that the size-report will be 2x longer if we simply add 1.2 results under 1.1 results. 

I think a better idea might be put 1.2 columns after current 1.1 columns:
- Remove the `total` column to save space for 3 more columns. (I think summing up `text` increase and `memory` increases does not make much sense. )
- Add `text`, `data`, `bss` columns for 1.2 to the right of 1.1 columns.

Reply:
Adding columns sounds like a good idea.

We now know that this PR adds 2048 bytes code and 32 bytes RAM. I think we should remove this flag in this PR and have a separate PR that updates size-report to include 1.2 changes.

Thoughts?

Reply:
Example:
# Size Report of **OpenThread**
> Merging [#5268](https://github.com/openthread/openthread/commit/7497659a1b11ef1aea9b6774c9f6e9239a4b7440) into [master](https://github.com/openthread/openthread/commit/57ee45be21c2b6cbba0dc53cea235fca33772cbc)(57ee45be21c2b6cbba0dc53cea235fca33772cbc).

|  name  |  branch  |  text  | data  | bss  | total | text  | data  | bss  | total |
| :----: | :------: | -----: | ----: | ---: | ----: |  ----: | ----: | ----: | ----: |
| ot-cli-ftd | master | 396720 | 736 | 72828 | 470284 | 396720 | 736 | 72828 | 470284 |
|  | #5268 | 396720 | 736 | 72828 | 470284 | 396720 | 736 | 72828 | 470284 |
|  | +/- | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
| ot-cli-mtd | master | 319360 | 736 | 64180 | 384276 | 319360 | 736 | 64180 | 384276 |
|  | #5268 | 319360 | 736 | 64180 | 384276 | 319360 | 736 | 64180 | 384276 |
|  | +/- | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |

Reply:
sgtm to add columns for 1.2 size. I may suggest to keep `total` as it is more visible to learn the overall consumption.
The other option is two reports separately for 1.1 and 1.2 if there is width concern/issue for the 'add columns' option

Reply:
@jwhui 
check-size change has been removed from this PR. Let's discuss this further in https://github.com/openthread/openthread/issues/5274.